specs/CustomDownload.md

```diff
@@ -0,0 +1,374 @@
+
+    ```c# (but really MIDL3)
+    public class CoreWebView2NewWindowRequestedEventArgs
+    {
+        // ...
+
+	       public CoreWebView2WindowFeatures WindowFeatures { get; }
+    }
+    ```
+-->
+## Win32 C++
+```c#
+[uuid(9EAFB7D0-88C3-4450-BBFB-C05A46C40C72), object, pointer_default(unique)]
+interface ICoreWebView2_3 : ICoreWebView2_2 {
+
+  /// Add an event handler for the `DownloadStarting` event. This event is
+  /// raised when a download has begun, blocking the default download dialog,
+  /// but not blocking the progress of the download.
+  ///
+  /// The host can choose to cancel a download, change the result file path, and hide
+  /// the default download dialog. If the host chooses to cancel the download, the
+  /// download is not saved and no dialog is shown. Otherwise, the download is saved
+  /// once the event completes, and default download dialog is shown if the host
+  /// did not choose to hide it.
+  ///
+  /// If the event is not handled, downloads will complete normally with the
+  /// default dialog shown.
+  ///
+  HRESULT add_DownloadStarting(
+    [in] ICoreWebView2DownloadStartingEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_DownloadStarting`.
+  HRESULT remove_DownloadStarting(
+      [in] EventRegistrationToken token);
+}
+
+[uuid(53b676ec-c3b1-4804-996a-c72c16b09efb), object, pointer_default(unique)]
+interface ICoreWebView2DownloadStartingEventHandler : IUnknown
+{
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2_3* sender,
+      [in] ICoreWebView2DownloadStartingEventArgs* args);
+}
+
+/// Event args for the `DownloadStarting` event.
+[uuid(c7e95e2f-f789-44e6-b372-3141469240e4), object, pointer_default(unique)]
+interface ICoreWebView2DownloadStartingEventArgs : IUnknown
+{
+  /// The URI of the download.
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The Content-Disposition header value from HTTP response.
+  [propget] HRESULT ContentDisposition([out, retval] LPWSTR* contentDisposition);
+
+  /// MIME type of the downloaded content.
+  [propget] HRESULT MimeType([out, retval] LPWSTR* mimeType);
+
+  /// The size of the download in total number of expected bytes.
+  [propget] HRESULT DownloadSizeInBytes([out, retval] UINT64* downloadSizeInBytes);
+
+  /// The host may set this flag to cancel the download. If canceled, the
+  /// download save dialog will not be displayed regardless of the
+  /// ShouldDisplayDefaultDownloadDialog property.
+  [propget] HRESULT Cancel([out, retval] BOOL* cancel);
+
+  /// Sets the `Cancel` property.
+  [propput] HRESULT Cancel([in] BOOL cancel);
+
+  /// The path to the file. If setting the path, the host should ensure that it
+  /// is an absolute path, including the file name. If the directory does not
+  /// exist, it will be created.
+  [propget] HRESULT ResultFilePath([out, retval] LPWSTR* resultFilePath);
```

Comment:
What happens when this path points to an existing file? Will the file be overwritten or will it receive a suffix?